docs/system_overview/process_model.md

```diff
@@ -1,29 +1,36 @@
+- **Agent Factory**: Handles all incoming API requests
+- **Delay Server** (optional): Manages delayed rule execution
 
-    /usr/sbin/irodsServer
-    ├─ irodsDelayServer
-    └─ irodsAgentFactory
-       ├─ irodsAgent
-       ├─ irodsAgent
-       ├─ irodsAgent
-       └─ irodsAgent
+## How requests are handled
 
-| iRODS Process         | Expected Duration        | Expected Resident Memory Usage  |
-| --------------------- | ------------------------ | ------------------------------- |
-| irodsServer           | long-running             | 126M                            |
-| irodsDelayServer      | long-running             | 17M                             |
-| irodsAgentFactory     | long-running             | 5M                              |
-| irodsAgent            | single client connection | 20M to 100M                     |
+The **Agent Factory** listens for new client connections. For each new connection, it forks a separate **Agent** process. Each Agent handles one client request, completes the work as quickly as possible, and then waits for the next request. Agents will continue to service requests until the client disconnects.
 
-iRODS uses a dynamic linking model, rather than a static linking model, which adds significant overhead for short running processes (e.g. `ils`).  In order to mitigate this overhead, the irodsServer no longer calls `fork()` and `exec()` as a separate executable (pre-4.2).  In addition, since calling `fork()` within a threaded environment creates undefined behavior, the irodsServer calls `fork()` to spawn the Agent Factory process early, before any threading begins and any configuration is determined.
+## The Delay Server
 
-![iRODS Process Model Diagram](../images/process_model_diagram.jpg)
+The **Delay Server** is designed to remain idle most of the time. Every 30 seconds _(by default)_, it wakes up and spawns an Agent to check for any delayed rules that need to run. You can adjust this interval using the `delay_server_sleep_time_in_seconds` property in server_config.json. See [Configuration](../system_overview/configuration.md#etcirodsserver_configjson) for more information about this property.
 
+## Example Process Tree
 
+    irodsServer
+      ├─irodsAgent
+      │   ├─irodsAgent
+      │   ├─irodsAgent
+      │   └─irodsAgent
+      └─irodsDelayServer
 
+| iRODS Process         | Expected Duration        | Expected Memory Usage           |
+| --------------------- | ------------------------ | ------------------------------- |
+| irodsServer           | Long-running             | ~34 MB                          |
+| irodsDelayServer      | Long-running             | ~33 MB                          |
+| irodsAgent (factory)  | Long-running             | ~92 MB                          |
+| irodsAgent            | Depends on client        | ~92 MB (minimum)                |
+
+TODO: Update or remove graphic.
+![iRODS Process Model Diagram](../images/process_model_diagram.jpg)
```

Comment:
My initial thought is that the process model diagram can be removed.

@trel Thoughts?

Reply:
i looked at it the other day - it doesn't seem ... wrong?     what about it is no longer true?

Reply:
The parts which mention `execAgent`, `socket filename`, `params` are no longer true.

The main reason is that I don't think it adds much value to the docs.

Reply:
if there are parts that are wrong/misleading - then certainly we don't want that.

i think with the text rendering just above... the image could be removed, yes.

Reply:
Agreed.

Reply:
Removed image.